sklearn/cluster/optics_.py

```diff
@@ -85,41 +86,19 @@ class OPTICS(BaseEstimator, ClusterMixin):
     metric_params : dict, optional (default=None)
         Additional keyword arguments for the metric function.
 
-    maxima_ratio : float, optional (default=.75)
-        The maximum ratio we allow of average height of clusters on the
-        right and left to the local maxima in question. The higher the
-        ratio, the more generous the algorithm is to preserving local
-        minima, and the more cuts the resulting tree will have.
-
-    rejection_ratio : float, optional (default=.7)
-        Adjusts the fitness of the clustering. When the maxima_ratio is
-        exceeded, determine which of the clusters to the left and right to
-        reject based on rejection_ratio. Higher values will result in points
-        being more readily classified as noise; conversely, lower values will
-        result in more points being clustered.
-
-    similarity_threshold : float, optional (default=.4)
-        Used to check if nodes can be moved up one level, that is, if the
-        new cluster created is too "similar" to its parent, given the
-        similarity threshold. Similarity can be determined by 1) the size
-        of the new cluster relative to the size of the parent node or
-        2) the average of the reachability values of the new cluster
-        relative to the average of the reachability values of the parent
-        node. A lower value for the similarity threshold means less levels
-        in the tree.
-
-    significant_min : float, optional (default=.003)
-        Sets a lower threshold on how small a significant maxima can be.
+    cluster_method : string, optional (default='dbscan')
+        The extraction method used to extract clusters using the calculated
+        reachability and ordering. Possible values are "dbscan".
+
+    eps : float, optional (default=0.5)
+        The maximum distance between two samples for them to be considered
+        as in the same neighborhood. Used ony when `cluster_method='dbscan'`.
```

Comment:
I think the parameter name `eps` is confusing if it is not the `eps` in the OPTICS paper (which is `max_eps` above - the name max eps is IMHO a good idea, but maybe call this `dbscan_eps`?). Also, if `eps` is set, we should use `max_eps=eps` for efficiency.
But I also see less use in having `cluster_method='dbscan'` since there is a separate DBSCAN in sklearn anyway.
Assuming that the next step is to merge the xi method, and make it default, the `eps` parameter may become more confusing than useful.

Reply:
This is different if the cluster extraction would be a separate step.
It makes a lot of sense to run OPTICS with a `max_eps` only once and the extract DBSCAN clusters multiple times for different `eps` values. But as far as I can tell, this is now not possible anymore (this probably should be a public method `extract_dbscan`, and only this method should have the `eps` parameter).

Reply:
there's another issue related to this, which is to have a `warm_start` exactly for this reason.

Also, the default `cluster_method` would be `xi` after the other PR is merged.

Another point is that for some reason we expect the `max_eps` to be at least `5*eps` for dbscan extraction work similarly to a usual dbscan.

Reply:
If this `max_eps >= 5*eps` thing is a reasonable expectation, then I'm happy to have it as `dbscan_eps`... but yes it's a bit obscure if as long as we do not support `warm_start`/`memory` or have a `extract(method=blah)` method